routing extension additions break CESM driver compiling

When using the current `develop` branch, I get an error while compiling the CESM driver because the `rout_extension_name.h` file cannot be found: 

```In file included from /p/home/gergel/rasm_vic5/models/lnd/vic/vic/drivers/shared_image/src/alloc_atmos.c(27):
/p/home/gergel/rasm_vic5/models/lnd/vic/vic/drivers/shared_image/include/vic_driver_shared_image.h(30): catastrophic error: cannot open source file "rout_extension_name.h"
  #include <rout_extension_name.h>
                                  ^

compilation aborted for /p/home/gergel/rasm_vic5/models/lnd/vic/vic/drivers/shared_image/src/alloc_atmos.c (code 4)
gmake: *** [alloc_atmos.o] Error 4
gmake: *** Waiting for unfinished jobs....
In file included from /p/home/gergel/rasm_vic5/models/lnd/vic/vic/drivers/shared_image/src/alloc_veg_hist.c(27):
/p/home/gergel/rasm_vic5/models/lnd/vic/vic/drivers/shared_image/include/vic_driver_shared_image.h(30): catastrophic error: cannot open source file "rout_extension_name.h"
  #include <rout_extension_name.h>
                                  ^

compilation aborted for /p/home/gergel/rasm_vic5/models/lnd/vic/vic/drivers/shared_image/src/alloc_veg_hist.c (code 4)
gmake: *** [alloc_veg_hist.o] Error 4```

The `~/extensions` paths need to be added to the CESM driver build routines so that they are able to be found at compile-time.

--- a/vic/drivers/cesm/bld/vic.buildexe.csh
+++ b/vic/drivers/cesm/bld/vic.buildexe.csh
@@ -3,7 +3,7 @@
 cd $OBJROOT/lnd/obj
 
 set comp = 'unknown'
-set rout_rvic = 'rout_stub'
+set ROUT = 'rout_stub'
 
 if ($COMP_INTERFACE == 'MCT' ) set comp = mct
 if ($COMP_INTERFACE == 'ESMF') set comp = esmf
@@ -22,8 +22,6 @@ $CODEROOT/lnd/vic/vic/drivers/cesm/include
 $CODEROOT/lnd/vic/vic/drivers/cesm/cpl_$comp/
 $CODEROOT/lnd/vic/vic/extensions/rout_stub/src
 $CODEROOT/lnd/vic/vic/extensions/rout_stub/include
-$CODEROOT/lnd/vic/vic/extensions/$rout_rvic/src
-$CODEROOT/lnd/vic/vic/extensions/$rout_rvic/include
 EOF
 
 set vicdefs = ""

--- a/vic/drivers/cesm/bld/vic.buildexe.csh
+++ b/vic/drivers/cesm/bld/vic.buildexe.csh
@@ -3,6 +3,8 @@
 cd $OBJROOT/lnd/obj
 
 set comp = 'unknown'
+set rout_rvic = 'rout_stub'
+
 if ($COMP_INTERFACE == 'MCT' ) set comp = mct
 if ($COMP_INTERFACE == 'ESMF') set comp = esmf
 
@@ -18,6 +20,10 @@ $CODEROOT/lnd/vic/vic/drivers/shared_image/include
 $CODEROOT/lnd/vic/vic/drivers/cesm/src
 $CODEROOT/lnd/vic/vic/drivers/cesm/include
 $CODEROOT/lnd/vic/vic/drivers/cesm/cpl_$comp/
+$CODEROOT/lnd/vic/vic/extensions/rout_stub/src
+$CODEROOT/lnd/vic/vic/extensions/rout_stub/include
+$CODEROOT/lnd/vic/vic/extensions/$rout_rvic/src
+$CODEROOT/lnd/vic/vic/extensions/$rout_rvic/include
 EOF
 
 set vicdefs = ""

--- a/vic/extensions/rout_stub/src/rout.c
+++ b/vic/extensions/rout_stub/src/rout.c
@@ -32,13 +32,6 @@
 void
 rout_alloc(void)
 {
-    extern rout_struct rout;
-
-    // We need to write at least something to the VIC state file (if enabled).
-    // Therefore we make the dimension size 1*1.
-    rout.rout_param.n_outlets = 1;
-    rout.rout_param.full_time_length = 1;
-    rout.ring[1] = 0;
 }
 
 /******************************************************************************

--- a/vic/extensions/rout_stub/src/rout.c
+++ b/vic/extensions/rout_stub/src/rout.c
@@ -32,6 +32,13 @@
 void
 rout_alloc(void)
 {
+    extern rout_struct rout;
+
+    // We need to write at least something to the VIC state file (if enabled).
+    // Therefore we make the dimension size 1*1.
+    rout.rout_param.n_outlets = 1;
+    rout.rout_param.full_time_length = 1;
+    rout.ring[1] = 0;
 }
 
 /******************************************************************************

